src/AzureRMTools.ts

```diff
@@ -1263,8 +1265,8 @@ export class AzureRMTools {
                 // Make sure the kind of item being renamed is valid
                 const result: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
                 const referenceSiteInfo: IReferenceSite | undefined = pc.getReferenceSiteInfo(true);
-                if (referenceSiteInfo && referenceSiteInfo.definition.definitionKind === DefinitionKind.BuiltinFunction) {
-                    throw new Error("Built-in functions cannot be renamed.");
+                if (referenceSiteInfo && canRename(referenceSiteInfo)) {
```

Comment:
It would be better not to call the function twice, two reasons: 1) perf (although not really an issue here), 2) duplicate code is a code smell.

Suggestion:

                const referenceSiteInfo: IReferenceSite | undefined = pc.getReferenceSiteInfo(true);
                let renameError = referenceSiteInfo && getRenameError(referenceSiteInfo);
                ...

Reply:
Fixed.